test/bio_read_test.c

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+#include <stdio.h>
+#include <string.h>
+#include <openssl/evp.h>
+#include <openssl/bio.h>
+#include <openssl/rand.h>
+#include <openssl/bio.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <unistd.h>
```

Comment:
Won't work on windows!

Reply:
If I have ifdef the entire test. Not sure what is the best way to test these things on windows.